tests/components/swiss_public_transport/test_config_flow.py

```diff
@@ -0,0 +1,295 @@
+"""Test the swiss_public_transport config flow."""
+from unittest.mock import AsyncMock, patch
+
+from opendata_transport.exceptions import (
+    OpendataTransportConnectionError,
+    OpendataTransportError,
+)
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.swiss_public_transport import config_flow
+from homeassistant.components.swiss_public_transport.const import (
+    CONF_DESTINATION,
+    CONF_START,
+)
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_flow_user_init_data_success(hass: HomeAssistant) -> None:
+    """Test success response."""
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user"}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["handler"] == "swiss_public_transport"
+    assert result["data_schema"] == config_flow.DATA_SCHEMA
+
+    with patch(
+        "homeassistant.components.swiss_public_transport.config_flow.OpendataTransport",
+        autospec=True,
+    ) as mock_OpendataTransport:
+        mock_OpendataTransport_instance = mock_OpendataTransport.return_value
+        mock_OpendataTransport_instance.async_get_data.return_value = True
```

Comment:
I think this can be shorter and a bit more readable. Can you maybe try this:

```suggestion
    with patch(
        "homeassistant.components.swiss_public_transport.config_flow.OpendataTransport.async_get_data",
    ):
```

This can require some tinkering, so feel free to hmu on Discord to have a chat about it